.gitignore

```diff
@@ -0,0 +1,6 @@
+**/test/*
```

Comment:
I'm pretty sure that the line `test/*` is the only thing you need to ignore the all folder.
So you can remove the others.